modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/Enrich.scala

```diff
@@ -106,79 +118,84 @@ object Enrich {
         etlTstamp <- Clock[F].realTime(TimeUnit.MILLISECONDS).map(millis => new DateTime(millis))
         registry <- enrichRegistry
         enriched <- EtlPipeline.processEvents[F](adapterRegistry, registry, igluClient, processor, etlTstamp, payload)
-        trackLatency = enrichLatency(collectorTstamp)
-      } yield Payload(enriched, trackLatency *> row.finalise)
+        _ <- enrichLatency(collectorTstamp)
```

Comment:
Does it mean that we're changing the way we track latency? Previously it was guaranteed that the latency is tracked after the payload is back in the queue and acked. Now it seems it will be happening earlier - once we enriched the event?

Reply:
You're right this is not correct, I need to move the latency to after the sinking.